### PR TITLE
change FB sign-in from public library/profile page

### PR DESCRIPTION
**Pull request by ahsu1230**
Change FB signup from public profile/library to match the behavior of twitter.
When you click on 'Continue with Facebook', the entire browser will change -> move to /signup/facebook
and use the `?redirect=___&intent=follow` to redirect back to the correct page after signup.

@andrewconner

### CODE REVIEW

**Comment by andrewconner** [comment]
Great, will review soon.

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 1]
```diff
@@ -65,8 +65,39 @@ angular.module('kifi')
...
+        path += $window.encodeURIComponent($scope.userData.redirectPath);
+      } else {
+        var currentPath = $location.url().split('?')[0]; //remove query params when redirecting back to this page
+        path += $window.encodeURIComponent(currentPath);
+      }
+
+      if ($scope.userData.intent === 'follow') {
+        path += '&intent=follow';
+      }
+      return path;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js]: How about keeping the logic here, but moving the URL formatting to `routeService`?

``` js
return routeService.socialSignup(
  network,
  $scope.userData.redirectPath || $location.path(),
  $scope.userData.intent);
```

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 2]
```diff
@@ -412,7 +412,8 @@ class AuthController @Inject() (
               if (cookieIntent.isDefined) {
                 initParams ++= s"&intent=${cookieIntent.get.value}"
               }
-              Redirect(s"${redirectPath}${initParams.toString}")
+              val discardedCookies = Seq(cookieRedirect, cookieIntent).flatten.map(c => DiscardingCookie(c.name))
+              Redirect(s"${redirectPath}${initParams.toString}").discardingCookies(discardedCookies: _*)
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:416]: discarding cookies after using `redirect` and `intent`

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala [hunk 3]
```diff
@@ -216,13 +216,14 @@ class AuthHelper @Inject() (
...
+          val redirectPath = java.net.URLDecoder.decode(cookieRedirect.get.value, "UTF-8")
           val initParams = new StringBuilder("?install=1") // tell browser to pop-up install modal
           if (cookieIntent.isDefined) {
             initParams ++= s"&intent=${cookieIntent.get.value}"
           }
-          val returnUri = redirectUrl + initParams.toString
-          Ok(Json.obj("uri" -> returnUri))
+          val returnUri = redirectPath + initParams.toString
+          val discardedCookies = Seq(cookieRedirect, cookieIntent).flatten.map(c => DiscardingCookie(c.name))
+          Ok(Json.obj("uri" -> returnUri)).discardingCookies(discardedCookies: _*)
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:226]: discarding cookies after using `redirect` and `intent`

**Comment by 2is10** [comment]
LGTM w/latest changes. Thanks!

**Pull request merged**